cli/command/node/ps_test.go

```diff
@@ -0,0 +1,142 @@
+package node
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/swarm"
+	"github.com/docker/docker/pkg/testutil/assert"
+	"github.com/docker/docker/pkg/testutil/golden"
+)
+
+func TestNodePsErrors(t *testing.T) {
+	testCases := []struct {
+		args            []string
+		flags           map[string]string
+		infoFunc        func() (types.Info, error)
+		nodeInspectFunc func() (swarm.Node, []byte, error)
+		taskListFunc    func(options types.TaskListOptions) ([]swarm.Task, error)
+		taskInspectFunc func(taskID string) (swarm.Task, []byte, error)
+		expectedError   string
+	}{
+		{
+			infoFunc: func() (types.Info, error) {
+				return types.Info{}, fmt.Errorf("error asking for node info")
+			},
+			expectedError: "error asking for node info",
+		},
+		{
+			args: []string{"nodeID"},
+			nodeInspectFunc: func() (swarm.Node, []byte, error) {
+				return swarm.Node{}, []byte{}, fmt.Errorf("error inspecting the node")
+			},
+			expectedError: "error inspecting the node",
+		},
+		{
+			args: []string{"nodeID"},
+			taskListFunc: func(options types.TaskListOptions) ([]swarm.Task, error) {
+				return []swarm.Task{}, fmt.Errorf("error returning the task list")
+			},
+			expectedError: "error returning the task list",
+		},
+		/*
+			{
+				args: []string{"nodeID"},
+				taskListFunc: func(options types.TaskListOptions) ([]swarm.Task, error) {
+					return []swarm.Task{
+						aTask("").build(),
+					}, nil
+				},
+				taskInspectFunc: func(taskID string) (swarm.Task, []byte, error) {
+					return swarm.Task{}, []byte{}, fmt.Errorf("error inspecting a task")
+				},
+				expectedError: "error returning the task list",
+			},
+		*/
```

Comment:
Yeah, related to the *should it be an error* question, I should remove it yeah 👍